src/IO/ReadHelpers.cpp

```diff
@@ -1140,6 +1150,8 @@ bool loadAtPosition(ReadBuffer & in, DB::Memory<> & memory, char * & current)
     if (current < in.buffer().end())
         return true;
 
+    while (!in.eof() && !in.hasPendingData()) {}
```

Comment:
This fixes case `input_format_parallel_parsing = 1`, but not sure it is appropriate place, will look at tests.